### PR TITLE
Hash check on the downloaded archives

### PR DESCRIPTION
**Pull request by guruDanny67**
resolv #25

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1653,14 +1715,34 @@ def __copy(self, src, destdir):
...
             shutil.copy(src, destdir)
 
+    def __hashfile(self, file_name):
+        hash_calc = hashlib.sha256()
+        with open(file_name, 'rb') as fi:
+            for chunk in iter(lambda: fi.read(4096), b""):
+                hash_calc.update(chunk)
+        return hash_calc.hexdigest()
+
+    def __check_hash(self, proj):
```
Comment (nacho) [inline build.py]: Let's return true or false insetad of 1 or 0

File: build.py [hunk 2]
```diff
@@ -1653,14 +1715,34 @@ def __copy(self, src, destdir):
...
+            hc = self.__hashfile(proj.archive_file)
+            if hc != proj.hash:
+                print_message("Hash mismatch on %s:\n  Calculated '%s'\n  Expected   '%s'\n" % (proj.archive_file, hc, proj.hash, ))
+                return 1
+            else:
+                if self.opts.check_hash:
+                    print_message("Hash ok on %s (%s)" % (proj.archive_file, hc, ))
+                else:
+                    print_debug("Hash ok on %s (%s)" % (proj.archive_file, hc, ))
+        return 0
```
Comment (nacho) [inline build.py]: same here, return True or False

File: build.py [hunk 3]
```diff
@@ -1653,14 +1715,34 @@ def __copy(self, src, destdir):
...
+
     def __download_one(self, proj):
         if not proj.archive_file:
             print_debug("archive file is not specified for project %s, skipping" % (proj.name,))
-            return
+            return 0
```
Comment (nacho) [inline build.py]: return False

Comment (guruDanny67) [inline build.py]: For the last changes I delete & redo the pull request. Now I only make the change, pull on my repository and make/update a pull request: there is no confirmation but it seem that the change is now visible to you. Is correct or I have to do something more ?

Thanks.

File: build.py (hunk 1 above)
Comment (guruDanny67) [inline build.py]: Done. Thanks

File: build.py [hunk 4]
```diff
@@ -1653,14 +1715,34 @@ def __copy(self, src, destdir):
...
+                hash_calc.update(chunk)
+        return hash_calc.hexdigest()
+
+    def __check_hash(self, proj):
+        if hasattr(proj, 'hash'):
+            hc = self.__hashfile(proj.archive_file)
+            if hc != proj.hash:
+                print_message("Hash mismatch on %s:\n  Calculated '%s'\n  Expected   '%s'\n" % (proj.archive_file, hc, proj.hash, ))
+                return True
+            else:
```
Comment (nacho) [inline build.py]: no need for this else, you are returning anyway

**Comment by guruDanny67** [comment]
Drop the else (but I leave the print of the correct hash, right ? I use it the first time to set the hashes :)

Thanks

**Comment by nacho** [comment]
Just do something like:
if pipo:
    return pepe

do_something_else

**Comment by guruDanny67** [comment]
Done, even if it's not very clear from the commit diff (but clearear on the global diff).

Thanks

**Comment by guruDanny67** [comment]
Good morning.

The pull request is still marked as 'changes requested' but I have done the modification some days ago: it's ok or I need to change something else ?

Thanks in advance

    Daniele Forghieri

**Comment by nacho** [comment]
See that there are conflicts, can u rebase your branch on top of master and fix the conflicts?

**Comment by guruDanny67** [comment]
Hi, now there is no more conflicts and the pull request should merge without problems even if I'm not really shure about the lz4 hash (in my branch I've got already the 1.7.4.2 version and not the new 1.7.5).

Let me know if it's ok and many thanks for your patience.

**Comment by nacho** [comment]
This seems wrong. I see lots of commits while there should be only one commit. You should squash your commit into one single one without doing a merge but rebasing your branch. I.e doing git rebase -i origin/master

**Comment by guruDanny67** [comment]
As I said in the other pull request I mess up my repository so it's easier (even if I know is less correct) to make another, clean, pull request with only one commit.

Many thanks

**Comment by nacho** [comment]
Closing this since it was already merged the other pull reuqest

**Pull request closed without merging**